improver_tests/nbhood/circular_kernel/test_CircularNeighbourhood.py

```diff
@@ -339,52 +305,50 @@ def test_single_point_on_corner(self):
         )
         for index, slice_ in enumerate(expected_centroid):
             expected[index][0:3] = slice_
-        ranges = 3
-        result = CircularNeighbourhood(weighted_mode=True).apply_circular_kernel(
-            self.cube, ranges
-        )
+        radius = 6000
+        result = Neighbourhood("circular", weighted_mode=True).run(self.cube, radius)
         self.assertArrayAlmostEqual(result.data, expected)
 
 
 class Test_run(IrisTest):
 
-    """Test the run method on the CircularNeighbourhood class."""
+    """Test the run method on the Neighbourhood class."""
 
-    RADIUS = 6100
+    RADIUS = 4000
 
     def test_basic(self):
         """Test that a cube with correct data is produced by the run method"""
 
         expected_data = np.array(
             [
-                [0.992, 0.968, 0.96, 0.968, 0.992],
-                [0.968, 0.944, 0.936, 0.944, 0.968],
-                [0.96, 0.936, 0.928, 0.936, 0.96],
-                [0.968, 0.944, 0.936, 0.944, 0.968],
-                [0.992, 0.968, 0.96, 0.968, 0.992],
+                [1.0, 1.0, 1.0, 1.0, 1.0],
+                [1.0, 0.916666, 0.875, 0.916666, 1.0],
+                [1.0, 0.875, 0.833333, 0.875, 1.0],
+                [1.0, 0.916666, 0.875, 0.916666, 1.0],
+                [1.0, 1.0, 1.0, 1.0, 1.0],
             ]
         )
 
         data = np.ones((5, 5), dtype=np.float32)
         data[2, 2] = 0
         cube = set_up_variable_cube(data, spatial_grid="equalarea",)
 
-        result = CircularNeighbourhood().run(cube, self.RADIUS)
+        result = Neighbourhood("circular",).run(cube, self.RADIUS)
         self.assertIsInstance(cube, Cube)
         self.assertArrayAlmostEqual(result.data, expected_data)
 
     def test_mask_cube(self):
-        """Test that a NotImplementedError is raised, if a mask cube is passed
-        in when using a circular neighbourhood, as this option is not
-        supported."""
+        """Test that a cube with the correct data and mask is produced
+        if the input cube is masked"""
 
         data = np.ones((5, 5), dtype=np.float32)
         data[2, 2] = 0
         cube = set_up_variable_cube(data, spatial_grid="equalarea",)
 
-        msg = "The use of a mask cube with a circular kernel is " "not yet implemented."
-        with self.assertRaisesRegex(NotImplementedError, msg):
-            CircularNeighbourhood().run(cube, self.RADIUS, mask_cube=cube)
+        result = Neighbourhood("circular",).run(cube, self.RADIUS, mask_cube=cube)
+        self.assertIsInstance(cube, Cube)
+        self.assertArrayAlmostEqual(result.data.data, np.ones((5, 5)))
+        self.assertArrayEqual(result.data.mask, ~data.astype(np.bool))
```

Comment:
I think that you only need `bool` here. I think using `np.bool` is deprecated, see #1646.

Reply:
Done